src/components/sandbox-menu/sandbox-menu.tsx

```diff
@@ -63,16 +77,12 @@ function saveConfig(baseUrl: string, config: Record<string, unknown>) {
   // compute result based on baseline and user values
   // TODO?: could be better, although bigger changes might require changes in config handling
   // also, currently no support for setting required/suggested
-  const baseline = getStore().get(EXTENSION_STATE_REDUCER_ATOM).configuration
-    .state;
-  const baselineEntries: [string, unknown][] = Object.entries(baseline)
-    .filter(([url]) => url.startsWith(baseUrl))
-    .map(([key, value]) => [key, value.modifications.value.content]);
-
-  const finalConfigEntries: Record<string, unknown> = {};
-  baselineEntries.forEach(([key, value]) => {
-    finalConfigEntries[key] = value;
-  });
+  const baseline = getStore().get(CONFIGURATION_DEFAULTS_REDUCER_ATOM);
```

Comment:
Ok, now I am confused. The "defaults" are the baseline?  
I thought defaults are the values provided by the options and since the "full" configuration is edited, it would aslo require the extensions states?

Reply:
There are three sources of config:
1. Values provided by options.yml files
2. Values provided by config.yml files
3. Values provided by the user

The confusing part is that `CONFIGURATION_DEFAULTS_REDUCER_ATOM` has taken the role of representing the first two sources (see reference A). It became that way because:
1. the Customisations tab doesn't need the separation between 1 and 2 (it only displays the full config, and only needs the defaults for displaying disabling UI elements if they use `enabled: url == value`, actually maybe it doesn't need the defaults... I will make an issue) see reference B), 
2. and (reference C) the "reset this option" button uses this state to quickly go from user customisations back to what the activate extensions provide (1) and demand (2).

Maybe it is time to rename all of them :)

To clarify using set theory, configs are key value pairs.
A. Values provided by options.yml files.
B. Values provided by config.yml files.
C. Values provided by the user.

A contains all keys. B contains a subset of A (or all keys of A). C contains a subset of A (or all keys of A).
The full config is constructed by taking A, then applying the key value pairs of B, and then applying the key value pairs of C.

The full config therefore contains the same keys as A.


Code References:

A:
https://github.com/UnofficialCrusaderPatch/UCP3-GUI/blob/c6b5e70c4922c8cf04ae6c2c3986adc3c0ec58c8/src/function/extensions/state/change.tsx#L58-L74

B:
https://github.com/UnofficialCrusaderPatch/UCP3-GUI/blob/c6b5e70c4922c8cf04ae6c2c3986adc3c0ec58c8/src/components/ucp-tabs/config-editor/ui-elements/ui-factory/CreateNumberInput.tsx#L52-L56

https://github.com/UnofficialCrusaderPatch/UCP3-GUI/blob/c6b5e70c4922c8cf04ae6c2c3986adc3c0ec58c8/src/function/extensions/state/change.tsx#L64-L69


C:
https://github.com/UnofficialCrusaderPatch/UCP3-GUI/blob/c6b5e70c4922c8cf04ae6c2c3986adc3c0ec58c8/src/components/ucp-tabs/config-editor/ui-elements/ui-factory/popover/ConfigPopover.tsx#L162-L165

Reply:
Ok, then ok for this issue. Renaming could definitly help.  
And I am all for refactoring the config keeping (you already know that I would be in favour of saving every config in its own object and just providing an interface to the rest of the API which performs caches and resolves on demand, maybe even with reactive changes per value, but this would likely be a lot of work).